From 9688ccc0874c0c513b73d01679b1b426f463477f Mon Sep 17 00:00:00 2001
From: Scott Talbert <swt@techie.net>
Date: Fri, 31 Mar 2023 18:04:26 -0400
Subject: [PATCH] WebView tests: Fix Selection test with WebKit 2.40+

For some reason, calling HasSelection() immediately after SelectAll()
causes the web extension to get stuck under WebKit 2.40.  Instead of
sleeping, call wxYield() a few times to let the event loop run a bit.
---
 tests/controls/webtest.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/tests/controls/webtest.cpp b/tests/controls/webtest.cpp
index c0eb77f52929..98a817c9971a 100644
--- a/tests/controls/webtest.cpp
+++ b/tests/controls/webtest.cpp
@@ -232,9 +232,11 @@ TEST_CASE_METHOD(WebViewTestCase, "WebView", "[wxWebView]")
         // With WebKit SelectAll() sends a request to perform the selection to
         // another process via proxy and there doesn't seem to be any way to
         // wait until this request is actually handled, so loop here for some a
-        // bit before giving up.
-        for ( wxStopWatch sw; !m_browser->HasSelection() && sw.Time() < 50; )
-            wxMilliSleep(1);
+        // bit before giving up.  Avoid calling HasSelection() right away
+        // without wxYielding a bit because this seems to cause the extension
+        // to hang with webkit 2.40.0+.
+        for ( wxStopWatch sw; sw.Time() < 50; )
+            wxYield();
 #endif // wxUSE_WEBVIEW_WEBKIT2
 
         CHECK(m_browser->HasSelection());
